File: hanoi.c
/*
 *
 * Towers of Hanoi.
 * n = 2 => ac, ab, cb
 */
#include <stdio.h>

void movesDisks(int n, char a, char b, char c) {

     if(n == 1) {

     	printf("(%c -> %c)", a, b); 

     } else {

         movesDisks(n - 1, a, c, b);

         movesDisks(1, a, b, c);

         movesDisks(n - 1, c, b, a);  

     }    

};

int main(int argc, char const *argv[])
{

    movesDisks(2, 'a','b','c'); 

    printf("\n");

	return 0;
}

File: hanoi.rb
# Towers of Hanoi
def Hanoi n, a, b, c

	if n == 1

	   print a ,"=>", b

	   print "\n"

	else	

	   Hanoi n - 1, a , c, b

	   Hanoi 1, a, b, c

	   Hanoi n - 1, c, b, a

    end		
   
end

Hanoi 3,'a','b','c'
